plugin-server/src/worker/plugins/loadPlugin.ts

```diff
@@ -70,22 +71,69 @@ export async function loadPlugin(server: Hub, pluginConfig: PluginConfig): Promi
                 return true
             } else {
                 pluginConfig.vm?.failInitialization!()
-                await processError(server, pluginConfig, `Could not load index.js for ${pluginDigest(plugin)}!`)
+                await processError(hub, pluginConfig, `Could not load index.js for ${pluginDigest(plugin)}!`)
+            }
+        } else if (plugin.plugin_type === 'source') {
+            if (
+                plugin.source_frontend &&
+                (plugin.transpiled_frontend === null || plugin.transpiled_frontend === undefined)
+            ) {
+                if (await getTranspilationLock(hub, plugin)) {
+                    const transpilationStartTimer = new Date()
+                    try {
+                        plugin.transpiled_frontend = transpileFrontend(plugin.source_frontend)
+                    } catch (error: any) {
+                        plugin.transpiled_frontend = "'error'"
+                        await processError(hub, pluginConfig, error)
+                        hub.statsd?.increment(`transpile_frontend.ERROR`, {
+                            plugin: plugin.name ?? '?',
+                            pluginId: `${plugin.id ?? '?'}`,
+                        })
+                    }
+                    hub.statsd?.timing(`transpile_frontend`, transpilationStartTimer, {
+                        plugin: plugin.name ?? '?',
+                        pluginId: `${plugin.id ?? '?'}`,
+                    })
+
+                    await hub.db.postgresQuery(
+                        'update posthog_plugin set transpiled_frontend = $1 where id = $2',
```

Comment:
Let's keep consistent
```suggestion
                        'UPDATE posthog_plugin SET transpiled_frontend = $1 WHERE id = $2',
```